gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/iceberg/IcebergRegisterStep.java

```diff
@@ -102,9 +117,19 @@ public void execute() throws IOException {
       String msg = "Destination table (with TableMetadata) does not exist: " + tnfe.getMessage();
       log.error(msg);
       throw new IOException(msg, tnfe);
-    } catch (ExecutionException | RetryException e) {
-      log.error("Exception Encountered while Registering Iceberg Table", e);
-      throw new RuntimeException(e);
+    } catch (ExecutionException executionException) {
+      String msg = "Failed to register iceberg table";
+      log.error(msg, executionException);
```

Comment:
given this is a heavily concurrent system and we regularly analyze log msgs perhaps across multiple runs, every log message must carry context.  for that reason, this class already has the members `srcTableIdStr` and `destTableIdStr`.  follow how the latter is used on L103